Cargo.toml

```diff
@@ -140,6 +140,7 @@ llvm-sys-160 = { package = "llvm-sys", version = "160.1.0", optional = true }
 once_cell = "1.16"
 parking_lot = "0.12"
 static-alloc = { version = "0.2", optional = true }
+trc = "1.2.2"
```

Comment:
What's the purpose of trc if Builder isn't usable in a multi threaded context and isn't clone?

Reply:
I used `Trc` because I needed a layer of indirection to modify the `PositionState` even if there is only a `&self`. Because I needed to use `Trc::as_ptr` and an unsafe block, I wanted to be sure about the intricacies of the implementation so that safety is upheld.

Reply:
It seems like we could just avoid the unsafety with `Cell` or `AtomicBool`

Reply:
True, I will implement that.